xdsl/transforms/DeadCodeElim.py

```diff
@@ -0,0 +1,21 @@
+from xdsl.traits import NoSideEffect
+from xdsl.pattern_rewriter import (PatternRewriter, PatternRewriteWalker,
+                                   RewritePattern, GreedyRewritePatternApplier)
+from xdsl.ir import MLContext, Operation
+from xdsl.dialects.builtin import ModuleOp
+
+
+class DCEImpl(RewritePattern):
```

Comment:
Why not spell out the name? If we go with the short name, I feel like we definitely need a doccomment

Reply:
I changed the name to `UnusedOperationRemover`, because that's what it actually is.